### PR TITLE
Updating the simple app template

### PR DESCRIPTION
**Pull request by AlfGot**
### Scope
The scope of this PR is starting a discussion about a possible improvement of the SimpleApp template, in order to include the ALB handling. The goal is deploying the ALB close to the application (when possible) from the infra sub stack

### Changes
- Adding the ALB creation in the infra substack
- Adding the ACM certificates creation in the Infra stack
- Adding the possibility to deploy Feature branches
- Provisioning an ALB for FeatureBranches through ALB controller and annotations
- Provisioning a listener rule for each feature branch

### CODE REVIEW

File: bases/simple-app/@{{ cookiecutter.project_slug }}/deploy/infra/variables.tf [hunk 1]
```diff
@@ -18,5 +29,16 @@ variable "github_repo" {
...
   description = "Git organization name"
   type        = string
-  default     = "dnd-it"
+  default     = "@{{ cookiecutter.github_org }}"
+}
+
+variable "acm_domain_names" {
+  description = "Main domain name"
+  type        = string
+  default     = ""
```
Comment (rbapst-tamedia) [inline bases/simple-app/@{{ cookiecutter.project_slug }}/deploy/infra/variables.tf:38]: I will remove the default. Not sure what will happen to create an ACM certificate with "" ?

File: bases/simple-app/@{{ cookiecutter.project_slug }}/deploy/infra/variables.tf [hunk 2]
```diff
@@ -18,5 +29,16 @@ variable "github_repo" {
...
+}
+variable "acm_alternative_domain_names" {
+  description = "list of alternative domain names"
+  type        = list(any)
```
Comment (rbapst-tamedia) [inline bases/simple-app/@{{ cookiecutter.project_slug }}/deploy/infra/variables.tf:42]:
```suggestion
  type        = list(string)
```

File: addons/github-workflows/@{{ cookiecutter.project_slug }}/.github/workflows/{% if cookiecutter.feature %}cleanup-feature-branch.yaml{% endif %} [hunk 3]
```diff
@@ -16,8 +16,8 @@ jobs:
     strategy:
       max-parallel: 1
       matrix:
-        stack: [app]
         {%- if cookiecutter.github_environments %}
+        stack: [app]
         environment: ['dev']
```
Comment (rbapst-tamedia) [inline addons/github-workflows/@{{ cookiecutter.project_slug }}/.github/workflows/{% if cookiecutter.feature %}cleanup-feature-branch.yaml{% endif %}:21]:
```suggestion
        environment: [dev]
```

**Review by rbapst-tamedia — COMMENTED** [review]
small changes.
otherwise it looks good.

File: bases/simple-app/@{{ cookiecutter.project_slug }}/.github/workflows/cleanup-feature-branch.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  pull-requests: write
+
+jobs:
+  destroy_feature:
+    name: Destroy Feature Branch
+    strategy:
+      max-parallel: 1
+      matrix:
+        stack: [app]
+        environment: ["dev"]
```
[the remark this answers is not in the document]
Comment (AlfGot) [reply bases/simple-app/@{{ cookiecutter.project_slug }}/.github/workflows/cleanup-feature-branch.yaml:20]: 👍

File: bases/simple-app/@{{ cookiecutter.project_slug }}/deploy/infra/alb.tf [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    name   = "tag:Name"
+    values = ["*public*"]
+  }
+}
+
+module "acm" {
+  source  = "terraform-aws-modules/acm/aws"
+  version = "5.1.1"
+
+  domain_name               = var.acm_domain_names
```
[the remark this answers is not in the document]
Comment (AlfGot) [reply bases/simple-app/@{{ cookiecutter.project_slug }}/deploy/infra/alb.tf:62]: 👍

**Comment by rbapst-tamedia** [comment]
I would not use "github_org" and "github_repo" terraform variables but directly put the @{{ cookiecutter.org }} and @{{ cookiecutter.repo }} variable in the terraform .tf files. What do you think ?

**Review by Davidoutz — APPROVED** [review]

**Pull request merged**